storage/range_command.go

```diff
@@ -337,21 +340,26 @@ func (r *Range) InternalRangeLookup(batch engine.Engine, args *proto.InternalRan
 			"Range lookup specified invalid maximum range count %d: must be > 0", rangeCount))
 		return nil
 	}
-	if args.IgnoreIntents {
-		rangeCount = 1 // simplify lookup because we may have to retry to read new
+	consistent := args.ReadConsistency != proto.INCONSISTENT
+	if consistent && args.IgnoreIntents {
+		reply.SetGoError(util.Errorf("can not read consistently and skip intents"))
+		return nil
+	}
+	if args.IgnoreIntents || consistent {
```

Comment:
Sorry - I hadn't seen that we've got a MaxRanges parameter in `InternalLookupRequest`. Will make the change you suggested and fix up a call site which I think doesn't specify it.